pkg/noderesourcetopology/filter.go

```diff
@@ -191,9 +195,6 @@ func (tm *TopologyMatch) Filter(ctx context.Context, cycleState *framework.Cycle
 	if nodeInfo.Node() == nil {
 		return framework.NewStatus(framework.Error, "node not found")
 	}
-	if v1qos.GetPodQOS(pod) == v1.PodQOSBestEffort {
-		return nil
-	}
```

Comment:
My 2 cents: pre-check if the Pod is "best efforts" is still the semantics we're pursuing - so that we do early return. It's just v1qos.GetPodQOS() is not the right function to use which only checks CPU and memory:

```go
var supportedQoSComputeResources = sets.NewString(string(core.ResourceCPU), string(core.ResourceMemory))
```

So WDYT on composing a "best-efforts++" function to filter out the pods this plugin doesn't care?

Reply:
Sure, sounds good. Instead of returning early if the pod belongs to `BestEffort` QoS, we need to return if pod belongs to `BestEffort` AND doesn't request devices (non-native resources). I will add a helper function to determine if the pod is requesting non-native resources.